boot/bootstate20.go

```diff
@@ -244,6 +253,50 @@ func (ks20 *bootState20Kernel) commit() error {
 	return nil
 }
 
+// chooseAndCommitSnapInitramfsMount chooses which snap should be mounted
+// during the early boot sequence, i.e. the initramfs, and commits that
+// choice if it needs state updated.
+// Choosing to boot/mount the base snap needs to be committed to the
+// modeenv, but no state needs to be committed when choosing to mount a
+// kernel snap.
+func (ks20 *bootState20Kernel) chooseAndCommitSnapInitramfsMount() (sn snap.PlaceInfo, err error) {
+	err = ks20.kModeenv.loadModeenv()
```

Comment:
the caller sets that in, do we need the load?

Reply:
hmm, I would rather have it to keep consistency with the other methods on bootState20Kernel that use the modeenv, but strictly speaking we probably don't need it right now

Reply:
I'm going to leave it in, but if someone feels strongly about this I can remove it